### ISSUE TITLE
About ImageNet pretrain

### ISSUE THREAD

**Issue by Sixkplus** [Megvii-BaseDetection/DynamicRouting#13]
Hi, Yanwei!

Thank you for your nice work!

I would like to modify the backbone and perform ImageNet pretraining but I was not experienced.
Can you share some hyperparameters (or some config files) with me, just like the following issues:
1. What does the classification head look like? (global avg pool + FC or just FC and how many layers)
2. Where should I put the CLS heads? Should I use put them for all the strides or which one of them?
3. Other hyperparameters like optimizer, LR, epochs, e.t.c.

Looking forward for your reply!

**Comment by yanwei-li** [comment]
Hi. Thanks for your interest. 
1. For classification, we **inverse the upsample module** at the end of the network (4 gray nodes in Fig. 2) to generate the feature map with **output stride 32**. Then, _a global average pooling combined with one FC layer_ is added for classification.
2. At the end of the network, please refer to 1.
3. Actually, we adopt a commonly used training strategy for ImageNet pretrain with _SGD_ optimizer,  _Lambda_ lr scheduler with a _initial LR 0.2_, _256 imgs/Batch for 120 Epoch_.

**Comment by Sixkplus** [comment]
@yanwei-li 

THX~

**Comment by Sixkplus** [comment]
While, Another question.
When perform pretraining on ImageNet, did you apply the **_drop_path_** scheme?

**Comment by Sixkplus** [comment]
While, Another question.
When perform pretraining on ImageNet, did you apply the drop_path scheme?

**Comment by Sixkplus** [comment]
While, Another question.
When perform pretraining on ImageNet, did you apply the drop_path scheme?

**Comment by yanwei-li** [comment]
As you can find in the related experiment `Seg.Layer16.ImageNet`, we do not use "drop path" for ImageNet pretrained model.

**Comment by Sixkplus** [comment]
> As you can find in the related experiment `Seg.Layer16.ImageNet`, we do not use "drop path" for ImageNet pretrained model.

Sorry for misunderstanding. I mean when perform training on ImageNet, did you apply the drop out?

**Comment by yanwei-li** [comment]
We do not apply the drop path schedule for ImageNet Pretraining.
